server/resp/src/main/java/org/infinispan/server/resp/operation/LCSOperation.java

```diff
@@ -0,0 +1,201 @@
+package org.infinispan.server.resp.operation;
+
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.server.resp.response.LCSResponse;
+
+public class LCSOperation {
+   private static final CompletionStage<LCSResponse> MISSING_ARGUMENTS = CompletableFuture
+         .failedFuture(new IllegalStateException("Missing arguments"));
+
+   public static CompletionStage<LCSResponse> performOperation(AdvancedCache<byte[], byte[]> cache,
+         List<byte[]> arguments) {
+      if (arguments.size() < 4)
+         return MISSING_ARGUMENTS;
+      LCSOperationContext lcsCtx = new LCSOperationContext(arguments);
+      lcsCtx.cache = cache;
+      return lcsCtx.cache.getAllAsync(Set.of(lcsCtx.key1, lcsCtx.key2))
+            .thenCompose((m) -> {
```

Comment:
Can just use `thenApply`, no need for thenCompose.

Reply:
Although I do wonder if in the future if we should optionally offload this to a blocking thread if the 2 Strings are very large.